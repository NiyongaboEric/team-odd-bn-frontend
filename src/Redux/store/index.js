import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/index';

const middleware = applyMiddleware(thunk);
const initialState = {};
const store = createStore(rootReducer, initialState, composeWithDevTools(middleware));

export default store;
