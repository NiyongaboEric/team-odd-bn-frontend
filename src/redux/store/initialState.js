const state = {
  auth: {
    signup: {
      signupInputData: {},
      signupData: { blockSpinner: 'blockSpinner' },
      signupError: { blockSpinner: 'blockSpinner' },
    },
    signin: {
      signinInputData: {},
      signinData: { blockSpinner: 'blockSpinner' },
      signinError: { blockSpinner: 'blockSpinner' },
    },
  },
  trips: {
    tripRequests: {
      getCity: [],
      onewayData: {},
      onewayError: {},
      onewayInput: {},
    },
    requests: {
      requestsData: [],
      singleRequestData: {},
      requestCommentsData: {},
      requestsError: {},
    },
  },
  viewProfile: {
    profile: {},
  },
  profileError: {},
};
export default state;
