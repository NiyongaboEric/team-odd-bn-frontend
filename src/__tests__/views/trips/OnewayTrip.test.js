import React from 'react';
import ReactDom from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import store from '../../../redux/store';
import OnewayTrip from '../../../containers/TripsContainer/oneway';
import initialState from '../../../redux/store/initialState';
import {
  init,
} from '../../../__mocks__/trips/requests';

const mockStore = configureStore([]);
let wrapper;
let stores;
describe('On Oneway page', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(
      <Provider store={store}>
        <Router>
          <OnewayTrip />
        </Router>
      </Provider>,
      div,
    );
  });
  it('Test snapshot using the initial state', () => {
    stores = mockStore(initialState);
    wrapper = mount(<Provider store={stores}><Router><OnewayTrip /></Router></Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
