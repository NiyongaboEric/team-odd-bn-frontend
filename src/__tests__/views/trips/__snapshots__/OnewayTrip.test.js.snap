// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On Oneway page Test snapshot using the initial state 1`] = `ReactWrapper {}`;
